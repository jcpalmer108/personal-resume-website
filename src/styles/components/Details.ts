import styled from "styled-components";

export const Wrapper = styled.div`
  padding-top: 35px;
  display: flex;
  flex-direction: column;
  row-gap: 10px;

  &:first-child {
    padding-top: 45px;
  }

`

export const Label = styled.div`
  font-family: Gilroy-Bold;
  font-size: 15px;
`

export const Info = styled.p`
  font-family: Gilroy-Regular;
  font-size: 12px;
  margin: 0px;
`