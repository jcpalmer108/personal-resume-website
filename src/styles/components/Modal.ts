import styled from "styled-components"
import { desktopBreakpoint, jet, off_black, silver_polish, tabletBreakpoint } from "../variables";

export const BackgroundOverlay = styled.div`
  position: fixed;
  top: 0px;
  background: black;
  height: 100%;
  width: 100%;
  z-index: 998;
  opacity: 0.8;
`

export const Window = styled.div`
  position: fixed;
  top: calc(15% / 2);
  left: calc(15% / 2);
  height: 85%;
  width: 85%;
  overflow: hidden;
  opacity: 1;
  z-index: 999;
  text-align: right;
`
export const CloseIcon = styled.img`
  width: 17px;
  padding-bottom: 5px;
`

export const InnerWindow = styled.div`
  background: white;
  height: calc(100% - 30px);
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;

`


export const LeftRightBorders = styled.div`
  width: 87%;
  border-left: solid 1px ${silver_polish};
  border-right: solid 1px ${silver_polish};
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const TopBottomBorders = styled.div`
  height: 100%;
  width: 100%;

  @media(min-width: ${desktopBreakpoint}px) {
    border-top: solid 1px ${silver_polish};
    border-bottom: solid 1px ${silver_polish};
    height: 80%;
  }

  `

export const Content = styled.div`
  text-align: left;
  padding: 20px;
  max-height: calc(100% - 40px);
  overflow-y: scroll;  

  @media(min-width: ${tabletBreakpoint}px) {
    padding: 40px;
    max-height: calc(100% - 80px);

  }

`

export const Tile = styled.div`
  width: 100%;
  height: 100%;
  background: ${off_black};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  padding-bottom: 25px;

  @media(min-width: ${tabletBreakpoint}px) {
    max-width: 235px;
    padding: 40px;
    margin: auto;
  }

`

export const Circle = styled.div`
  border-radius: 50%;
  background: ${jet};
  height: 145px;
  width: 145px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 35px;
  margin-bottom: 25px;

  @media(min-width: ${tabletBreakpoint}px) {
    height: 185px;
    width: 185px;
  }

`

type LogoProps = {
  icon?: string
}

export const Logo = styled.img<LogoProps>`
  height: ${({ icon }) => {
    switch(icon) {
      case "nike":
        return "28px";
      case "slalom": 
        return "50px";
      default: 
        return "70px";
    }
  }}};

  @media(min-width: ${tabletBreakpoint}px) {
    height: ${({ icon }) => {
      switch(icon) {
        case "nike":
          return "38px";
        case "slalom": 
          return "60px";
        default: 
          return "90px";
      }
    }}};  
  }

`

export const Overview = styled.div`
  display: flex;
  flex-direction: column;
  row-gap: 20px;

  @media(min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
    column-gap: 20px;
    padding-top: 75px;
  }

`

export const TileLabel = styled.div`
  font-family: Gilroy-Bold;
  font-size: 14px;

  @media(min-width: ${tabletBreakpoint}px) {
    font-size: 17px;
    padding-bottom: 15px;
  }

`

export const TileInfo = styled.span`
font-family: Gilroy-Regular;
font-size: 14px;

@media(min-width: ${tabletBreakpoint}px) {
  font-size: 13px;
}
`

export const OrgTeam = styled.div`
  display: flex;
  flex-direction: column;

  @media(min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
  }
`