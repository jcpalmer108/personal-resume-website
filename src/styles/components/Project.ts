import styled from "styled-components";
import { silver_polish, tabletBreakpoint } from "../variables";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;

  @media (min-width: ${tabletBreakpoint}px) {
    align-items: flex-start;
  }
`;

export const Icon = styled.img`
  height: 24px;
  width: 24px;
  border: 1px solid ${silver_polish};
  padding: 20px;
`;

export const Title = styled.div`
  font-family: Gilroy-Regular;
  font-size: 14px;
  font-weight: bold;
  padding-top: 15px;

  @media (min-width: ${tabletBreakpoint}px) {
    padding-top: 0px;
  }
`;
export const Industry = styled.div`
  font-family: Gilroy-Medium;
  color: ${silver_polish};
  font-size: 12px;
  padding-top: 10px;

  @media (min-width: ${tabletBreakpoint}px) {
    text-align: left;
  }
`;

export const Description = styled.div`
  text-align: left;
`;

export const Heading = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 25px;

  @media (min-width: ${tabletBreakpoint}px) {
    flex-direction: row;
    justify-content: flex-start;
    column-gap: 20px;
  }
`;
export const TitleLine = styled.div`
  align-text: left;
`;
