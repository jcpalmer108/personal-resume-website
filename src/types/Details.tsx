export type WrapperProps = {
  noBottom?: boolean | undefined;
}

export type DetailsProps = {
  label: string,
  info?: string[],
  noBottom?: boolean
}
