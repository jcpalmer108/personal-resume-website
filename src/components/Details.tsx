import { Wrapper, Label, Info } from "../styles/components/Details";
import { DetailsProps } from "../types/Details";

export default function Details({ label, info, noBottom }: DetailsProps) {
  if (!info) {
    return null;
  }

  return (
    <Wrapper noBottom={noBottom} data-testid="Details">
      <Label>{label}</Label>
      <div>
        {info.map((paragraph, index) => (
          <Info key={`Info ${index + 1}`}>{paragraph}</Info>
        ))}
      </div>
    </Wrapper>
  );
}
