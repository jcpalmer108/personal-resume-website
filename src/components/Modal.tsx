import closeIcon from "../assets/images/icons/close.svg";
import {
  BackgroundOverlay,
  CloseIcon,
  InnerWindow,
  LeftRightBorders,
  TopBottomBorders,
  Window,
} from "../styles/components/Modal";
import { ModalProps } from "../types/Modal";

export default function Modal({ closeModal, children }: ModalProps) {
  return (
    <>
      <BackgroundOverlay data-testid="Overlay" onClick={() => closeModal()} />
      <Window data-testid="Modal">
        <CloseIcon
          src={closeIcon}
          alt="close"
          data-testid="CloseIcon"
          onClick={() => closeModal()}
        />
        <InnerWindow>
          <LeftRightBorders>
            <TopBottomBorders data-testid="ChildrenWrapper">
              {children}
            </TopBottomBorders>
          </LeftRightBorders>
        </InnerWindow>
      </Window>
    </>
  );
}
