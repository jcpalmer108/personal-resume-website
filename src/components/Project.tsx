import { Description, Heading, Icon, Industry, Title, TitleLine, Wrapper } from "../styles/components/Project"
import { ProjectProps } from "../types/Project"

export default function Project({ icon, industry, title, description }: ProjectProps) {
  return (
    <Wrapper data-testid="Project">
      <Heading>
        <Icon data-testid="Icon" src={require("../assets/images/industries/" + icon + ".svg")} />
        <TitleLine>
          <Title data-testid="Title">{title}</Title>
          <Industry data-testid="Industry">{industry}</Industry>
        </TitleLine>
      </Heading>
      <Description data-testid="Description">
        {description.map((paragraph, index) => <p key={`Paragraph ${index + 1}`}>{paragraph}</p>)}
      </Description>
    </Wrapper>
  )
}