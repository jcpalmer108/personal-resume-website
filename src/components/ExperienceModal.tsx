import { Experience } from "../types/content"
import { 
  Content,
  Circle,
  Logo,
  Tile,
  Overview,
  TileLabel,
  TileInfo,
  OrgTeam,
  JobDetails,
  Summary
} from "../styles/components/Modal";
import Details from "./Details"
import Project from "./Project"
import Modal from "./Modal"

type ExperienceModalProps = {
  closeModal: Function
  content?: Experience;
}

export default function ExperienceModal({ closeModal, content }: ExperienceModalProps) {
  const generateDetails = () => {
    let response = [];

    if(content?.description) response.push(
      <Details 
        key="Description"
        data-testid="Description" 
        label="Description" 
        info={content.description}
      />
    )

    if(content?.organization && content?.team) response.push(
      <OrgTeam key="OrgTeam">
        <Details 
          label="Organization" 
          info={[content.organization]} 
          data-testid="Organization" 
        />
        <Details 
          label="Team" 
          info={[content.team]} 
          noBottom 
          data-testid="Team" 
        />
      </OrgTeam>
    )

    if(content?.location) response.push(
      <Details 
        key="Location"
        label="Location" 
        info={[content.location]}
        data-testid="Location" 
      />
    )

    if(content?.timeline) response.push(
      <Details 
        key="Timeline"
        label="Timeline" 
        info={[`${content.timeline.start} - ${content.timeline.end}`]} 
        data-testid="Timeline" 
      />
    )
    
    if(content?.projects) {
      response.push(<Details key="Projects" label="Projects" data-testid="Projects" />)
      response.push(
      <div data-testid="ProjectLineItems">
        {
          content.projects.map(({icon, industry, description, title}, index) => 
            <Project 
              key={`Project ${index + 1}`}
              icon={icon} 
              industry={industry} 
              description={description} 
              title={title} 
              data-testid={`Project ${index + 1}`}
            />
          )
        }
      </div>)
    }

    return response
  }

  return (
    <>
      <Modal closeModal={closeModal}>
        <Content data-testid="Content">
          <Tile>
            <Circle>
              <Logo data-testid="Logo" src={require(`../assets/images/${content?.icon || "nike"}-white.svg`)} alt="logo" icon={content?.icon} />
            </Circle>
            <Overview>
              <Summary>
                <TileLabel>Title</TileLabel>
                <TileInfo data-testid="Title">{content?.title}</TileInfo>
              </Summary>
              <Summary>
                <TileLabel>Employer</TileLabel>
                <TileInfo data-testid="Employer">{content?.employer}</TileInfo>
              </Summary>
            </Overview>
          </Tile>
          <JobDetails>
            {generateDetails().map((item, index) => <div key={`Detail ${index + 1}`}>{item}</div>)}
          </JobDetails>
        </Content>
      </Modal>
    </>
  )
}