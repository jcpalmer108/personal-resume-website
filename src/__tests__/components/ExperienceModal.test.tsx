import { render, screen } from "@testing-library/react";
import Modal from "../../components/Modal"
import { DetailsProps } from "../../types/Details";
import { ProjectProps } from "../../types/Project";
import { ModalProps } from "../../types/Modal";
import ExperienceModal from "../../components/ExperienceModal";

// mock components
const mockDetails = jest.fn();
jest.mock("../../components/Details", () => ({ label, info }: DetailsProps) => {
  mockDetails(label, info);
  return <div />;
});

const mockProject = jest.fn();
jest.mock("../../components/Project", () => ({ icon, industry, description, title }: ProjectProps) => {
  mockProject(icon, industry, description, title);
  return <div />;
});

const mockModal = jest.fn();
jest.mock("../../components/Modal", () => ({ closeModal, children }: ModalProps) => {
  mockModal(closeModal, children);
  return <div>{children}</div>;
});

// params needed for component
const params = {
  required: {
    closeModal: jest.fn(),
    content: {
      title: 'test title',
      employer: 'test employer',
      location: 'test location',
      description: [ '1', '2' ],
      icon: 'nike',
      timeline: {
        start: 'start date',
        end: 'end date'
      }  
    }
  },
  optional: {
    content: {
      team: 'test team',
      organization: 'test org',
      projects: [
        {
          icon: 'project icon 1',
          industry: 'project industry 1',
          description: [ 'project 1 desc 1', 'project 1 desc 2' ],
          title: 'project title 1'
        },
        {
          icon: 'project icon 2',
          industry: 'project industry 2',
          description: [ 'project 2 desc 1', 'project 2 desc 2' ],
          title: 'project title 2'
        }
      ]  
    }
  }
}

describe('ExperienceModal', () => {
  test('renders if optional and required params are passed in', () => {
    // given
    const { required, optional } = params;
    const completeContent = { ...required.content, ...optional.content }
    render(<ExperienceModal closeModal={required.closeModal} content={completeContent} />)

    // then
    expect(screen.getByTestId('ExperienceModal')).toMatchSnapshot()
    expect(screen.getByTestId("Title")).toHaveTextContent(required.content.title)
    expect(screen.getByTestId("Employer")).toHaveTextContent(required.content.employer)
    
    expect(mockDetails).toHaveBeenCalledTimes(6)
    expect(mockDetails).toHaveBeenNthCalledWith(1, "Description", required.content.description)
    expect(mockDetails).toHaveBeenNthCalledWith(2, "Organization", [optional.content.organization])
    expect(mockDetails).toHaveBeenNthCalledWith(3, "Team", [optional.content.team])
    
    expect(mockProject).toHaveBeenCalledTimes(2)
    new Array(2).forEach((index) => {
      expect(mockProject).toHaveBeenNthCalledWith(index + 1, 
        optional.content.projects[index].icon,
        optional.content.projects[index].industry,
        optional.content.projects[index].description,
        optional.content.projects[index].title
      )
    })
  })

  test('renders if only required params are passed in', () => {
    // given
    const { required } = params;
    render(<ExperienceModal closeModal={required.closeModal} content={required.content}/>)

    // then
    expect(screen.getByTestId('ExperienceModal')).toMatchSnapshot()
    expect(screen.getByTestId("Title")).toHaveTextContent(required.content.title)
    expect(screen.getByTestId("Employer")).toHaveTextContent(required.content.employer)
    
    expect(mockDetails).toHaveBeenCalledTimes(3)
    expect(mockDetails).toHaveBeenNthCalledWith(1, "Description", required.content.description)
    expect(mockDetails).toHaveBeenNthCalledWith(2, "Location", [required.content.location])
    expect(mockDetails).toHaveBeenNthCalledWith(3, "Timeline", [ `${required.content.timeline.start} - ${required.content.timeline.end}` ])

    expect(mockProject).toHaveBeenCalledTimes(0)
  })

})
