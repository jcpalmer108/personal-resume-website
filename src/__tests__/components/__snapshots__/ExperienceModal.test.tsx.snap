// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceModal renders if only required params are passed in 1`] = `
<div
  class="sc-gKXOVf cNOAyo"
  data-testid="ExperienceModal"
>
  <div
    class="sc-iBkjds lgAXJZ"
  >
    <div
      class="sc-ftvSup jPzHnb"
    >
      <img
        alt="logo"
        class="sc-papXJ ijcNsj"
        data-testid="Logo"
        src="[object Object]"
      />
    </div>
    <div
      class="sc-jqUVSM eukuru"
    >
      <div
        class="sc-kDDrLX erjzcT"
      >
        <div
          class="sc-iqcoie hZaRyn"
        >
          Title
        </div>
        <span
          class="sc-crXcEl YkJlI"
          data-testid="Title"
        >
          test title
        </span>
      </div>
      <div
        class="sc-kDDrLX erjzcT"
      >
        <div
          class="sc-iqcoie hZaRyn"
        >
          Employer
        </div>
        <span
          class="sc-crXcEl YkJlI"
          data-testid="Employer"
        >
          test employer
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-breuTD eUvivm"
  >
    <div />
    <div />
    <div />
  </div>
</div>
`;

exports[`ExperienceModal renders if optional and required params are passed in 1`] = `
<div
  class="sc-gKXOVf cNOAyo"
  data-testid="ExperienceModal"
>
  <div
    class="sc-iBkjds lgAXJZ"
  >
    <div
      class="sc-ftvSup jPzHnb"
    >
      <img
        alt="logo"
        class="sc-papXJ ijcNsj"
        data-testid="Logo"
        src="[object Object]"
      />
    </div>
    <div
      class="sc-jqUVSM eukuru"
    >
      <div
        class="sc-kDDrLX erjzcT"
      >
        <div
          class="sc-iqcoie hZaRyn"
        >
          Title
        </div>
        <span
          class="sc-crXcEl YkJlI"
          data-testid="Title"
        >
          test title
        </span>
      </div>
      <div
        class="sc-kDDrLX erjzcT"
      >
        <div
          class="sc-iqcoie hZaRyn"
        >
          Employer
        </div>
        <span
          class="sc-crXcEl YkJlI"
          data-testid="Employer"
        >
          test employer
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-breuTD eUvivm"
  >
    <div />
    <div
      class="sc-evZas jyOmOC"
    >
      <div />
      <div />
    </div>
    <div />
    <div />
    <div />
    <div
      data-testid="ProjectLineItems"
    >
      <div />
      <div />
    </div>
  </div>
</div>
`;
